sklearn/cluster/spectral.py

```diff
@@ -18,6 +21,41 @@
 from .k_means_ import k_means
 
 
+def clusterQR(vectors):
+    """Search for a partition matrix (clustering) which is
+    closest to the eigenvector embedding.
+
+    Parameters
+    ----------
+    vectors : array-like, shape: (n_samples, n_clusters)
+        The embedding space of the samples.
+
+    Returns
+    -------
+    labels : array of integers, shape: n_samples
+        The labels of the clusters.
+
+    References
+    ----------
+    https://github.com/asdamle/QR-spectral-clustering
+    https://arxiv.org/abs/1708.07481
+
+    Notes
+    -----
+    T.conj() allows the vectors to be complex-valued,
+    just in case for future use
+
+    """
+
+    k = vectors.shape[1]
+    piv = qr(vectors.T.conj(), pivoting=True)[2]
+    piv = piv[0:k]
+    Ut, Vt = svd(vectors[piv, :].T.conj())[0],\
```

Comment:
should not need `\`. Just assign Ut and Vt in separate statements

Reply:
in fact, why are we computing the SVD twice here?

Reply:
Great question! I no longer remember and need to re-investigate...

Reply:
Fixed both, thanks for noticing! The replacement lines are
```
UtSV = svd(vectors[piv, :].T.conj())
    Ut = UtSV[0]
    Vt = UtSV[2].T.conj()
```